### PR TITLE
Add primary variant to storefront product serializer

### PR DESCRIPTION
**Pull request by szymoniwacz**

### CODE REVIEW

File: api/app/serializers/spree/v2/storefront/product_serializer.rb [hunk 1]
```diff
@@ -59,11 +59,16 @@ class ProductSerializer < BaseSerializer
...
           serializer: :image
 
-        has_one  :default_variant,
+        has_one :default_variant,
           object_method_name: :default_variant,
           id_method_name: :default_variant_id,
           record_type: :variant,
           serializer: :variant
+
+        has_one :master,
```
Comment (damianlegawiec) [inline api/app/serializers/spree/v2/storefront/product_serializer.rb:68]: @szymoniwacz after 2nd thaught let's rename this to `master_variant`

Comment (damianlegawiec) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb:68]: @szymoniwacz line 9 - please add `sku` attribute there

Comment (damianlegawiec) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb:68]: I'm not happy with that `master` naming, which can be offensive in some cultures. 

https://www.theserverside.com/opinion/Master-slave-terminology-alternatives-you-can-use-right-now

`primary` is a good candidate to replace it, let's start with this serializer

File: api/spec/requests/spree/api/v2/resource_includes_spec.rb [hunk 2]
```diff
@@ -89,7 +92,7 @@
         end
 
         context 'with nested resources' do
-          before { get '/api/v2/storefront/products?include=option_types,option_types.option_values' }
+          before { get '/api/v2/storefront/products?include=master,option_types,option_types.option_values' }
```
Comment (damianlegawiec) [inline api/spec/requests/spree/api/v2/resource_includes_spec.rb:95]: @szymoniwacz let's also update Storefront Open API docs please

File: api/app/serializers/spree/v2/storefront/product_serializer.rb (hunk 1 above)
Comment (szymoniwacz) [reply api/app/serializers/spree/v2/storefront/product_serializer.rb:68]: @damianlegawiec 
I've added `:sku` in this PR: https://github.com/spree/spree/pull/11266/files.
You're right about _master_ naming.

**Review by damianlegawiec — APPROVED** [review]

File: api/app/serializers/spree/v2/storefront/product_serializer.rb [hunk 3]
```diff
@@ -54,16 +54,22 @@ class ProductSerializer < BaseSerializer
...
-          id_method_name: :variant_image_ids,
-          record_type: :image,
-          serializer: :image
-
-        has_one  :default_variant,
-          object_method_name: :default_variant,
-          id_method_name: :default_variant_id,
-          record_type: :variant,
-          serializer: :variant
+                 object_method_name: :variant_images,
```
Comment (damianlegawiec) [inline api/app/serializers/spree/v2/storefront/product_serializer.rb:57]: @szymoniwacz let's also do the same for platform serializer, please - https://github.com/spree/spree/blob/master/api/app/serializers/spree/api/v2/platform/product_serializer.rb#L47

**Pull request merged**